sdks/java/extensions/zetasketch/src/main/java/org/apache/beam/sdk/extensions/zetasketch/HllCount.java

```diff
@@ -0,0 +1,364 @@
+   * href="https://cloud.google.com/bigquery/docs/reference/standard-sql/hll_functions">BigQuery</a>.
+   */
+  public static final class MergePartial {
+
+    // Cannot be instantiated. This class is intended to be a namespace only.
+    private MergePartial() {}
+
+    /**
+     * Returns a {@link Combine.Globally} {@code PTransform} that takes an input {@code
+     * PCollection<byte[]>} of HLL++ sketches and returns a {@code PCollection<byte[]>} of a new
+     * sketch merged from the input sketches.
+     *
+     * <p>Only sketches of the same type and {@code precision} can be merged together. If
+     * incompatible sketches are provided, a runtime error will occur.
+     *
+     * <p>Returns an empty output {@code PCollection} if the input {@code PCollection} is empty.
+     */
+    public static Combine.Globally<byte[], byte[]> globally() {
+      return Combine.globally(HllCountMergePartialFn.create()).withoutDefaults();
+    }
+
+    /**
+     * Returns a {@link Combine.PerKey} {@code PTransform} that takes an input {@code
+     * PCollection<KV<K, byte[]>>} of (key, HLL++ sketch) pairs and returns a {@code
+     * PCollection<KV<K, byte[]>>} of (key, new sketch merged from the input sketches under the
+     * key).
+     *
+     * <p>Only sketches of the same type and {@code precision} can be merged together. If
+     * incompatible sketches are provided, a runtime error will occur.
+     */
+    public static <K> Combine.PerKey<K, byte[], byte[]> perKey() {
+      return Combine.perKey(HllCountMergePartialFn.create());
+    }
+  }
+
+  /**
+   * Provide {@code PTransform}s to extract the estimated count of distinct elements (as {@code
+   * Long}s) from each HLL++ sketch.
+   *
+   * <p>Correspond to the {@code HLL_COUNT.EXTRACT(sketch)} function in <a
+   * href="https://cloud.google.com/bigquery/docs/reference/standard-sql/hll_functions">BigQuery</a>.
+   */
+  public static final class Extract {
+
+    // Cannot be instantiated. This class is intended to be a namespace only.
+    private Extract() {}
+
+    /**
+     * Returns a {@code PTransform} that takes an input {@code PCollection<byte[]>} of HLL++
+     * sketches and returns a {@code PCollection<Long>} of the estimated count of distinct elements
+     * extracted from each sketch.
+     */
+    public static PTransform<PCollection<byte[]>, PCollection<Long>> globally() {
+      return new Globally();
+    }
+
+    /**
+     * Returns a {@code PTransform} that takes an input {@code PCollection<KV<K, byte[]>>} of (key,
+     * HLL++ sketch) pairs and returns a {@code PCollection<KV<K, Long>>} of (key, estimated count
+     * of distinct elements extracted from each sketch).
+     */
+    public static <K> PTransform<PCollection<KV<K, byte[]>>, PCollection<KV<K, Long>>> perKey() {
+      return new PerKey<K>();
+    }
+
+    private static final class Globally extends PTransform<PCollection<byte[]>, PCollection<Long>> {
```

Comment:
I would prefer keeping the classes where they are now. As you mentioned, `Extract` is a simple `PTransform`, so maybe not worth having its own implementation class (and technically the two `CombineFn`s for `Init` and `MergePartial` are not implementation of the `PTransform` itself neither). Although all the sub-classes are now under `HllCount`, they are disjoint and `HllCount` itself is just a stateless namespace, so I don't think it will harm readability much. Also this class layout is consistent with the other combine transforms in Beam (e.g. [Top](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Top.java)).